src/test/java/com/google/api/generator/engine/ast/NewObjectValueTest.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.api.generator.engine.ast;
+
+import static junit.framework.Assert.assertEquals;
+
+import java.awt.List;
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.Test;
+
+public class NewObjectValueTest {
+  @Test
+  public void simpleNewObjectValue() {
+    ConcreteReference reference =
+        ConcreteReference.builder()
+            .setClazz(List.class)
+            .setGenerics(Collections.emptyList())
+            .build();
+    TypeNode type = TypeNode.withReference(reference);
+    NewObjectValue newObjectValue = NewObjectValue.genericBuilder().setType(type).build();
+    assertEquals(newObjectValue.type(), type);
+    assertEquals(newObjectValue.value(), "new List<>");
+  }
+
+  @Test
+  public void newObjectValue_generics() {
+    ConcreteReference reference =
+        ConcreteReference.builder()
+            .setClazz(List.class)
+            .setGenerics(Arrays.asList(ConcreteReference.withClazz(String.class)))
+            .build();
+    TypeNode type = TypeNode.withReference(reference);
+    NewObjectValue newObjectValue = NewObjectValue.genericBuilder().setType(type).build();
+    assertEquals(newObjectValue.type(), type);
+    assertEquals(newObjectValue.value(), "new List<String>");
```

Comment:
If this were an ArrayList, would `new ArrayList<String>` be valid Java? Consider whether more stuff needs to be part of the returned `value()`.

Reply:
Rather, I meant that we should write the `()` all together. I think `NewObjectExpr` will allow us to do that, please see the comment above.